pkg/jobs/job_scheduler.go

```diff
@@ -375,6 +376,56 @@ func (s *jobScheduler) schedulerEnabledOnThisNode(ctx context.Context) bool {
 	return enabled
 }
 
+type syncCancelFunc struct {
+	syncutil.Mutex
+	context.CancelFunc
+	sv *settings.Values
```

Comment:
do we need to capture sv? don't we have it at the withCancelOnDisabled callsite too?

Reply:
yeah; we could pass it in...done.